api_app/connectors_manager/connectors/opencti.py

```diff
@@ -0,0 +1,199 @@
+        if self._job.is_sample:
+            obs_type = INTELOWL_OPENCTI_TYPE_MAP["file"]
+        elif self._job.observable_classification == "hash":
+            matched_hash_type = helpers.get_hash_type(self._job.observable_name)
+            if matched_hash_type in [
+                "md5",
+                "sha-1",
+                "sha-256",
+            ]:  # sha-512 not supported
+                obs_type = INTELOWL_OPENCTI_TYPE_MAP["file"]
+            else:
+                obs_type = INTELOWL_OPENCTI_TYPE_MAP["generic"]  # text
+        elif self._job.observable_classification == "ip":
+            ip_version = helpers.get_ip_version(self._job.observable_name)
+            if ip_version == 4 or ip_version == 6:
+                obs_type = INTELOWL_OPENCTI_TYPE_MAP["ip"][f"v{ip_version}"]  # v4/v6
+            else:
+                obs_type = INTELOWL_OPENCTI_TYPE_MAP["generic"]  # text
+        else:
+            obs_type = INTELOWL_OPENCTI_TYPE_MAP[self._job.observable_classification]
+
+        return obs_type
+
+    def generate_observable_data(self) -> dict:
+        observable_data = {"type": self.get_observable_type()}
+        if self._job.is_sample:
+            observable_data["name"] = self._job.file_name
+            observable_data["hashes"] = helpers.generate_hashes(self.job_id)
+        elif (
+            self._job.observable_classification == "hash"
+            and observable_data["type"] == "file"
+        ):
+            # add hash instead of value
+            matched_type = helpers.get_hash_type(self._job.observable_name)
+            observable_data["hashes"] = {matched_type: self._job.observable_name}
+        else:
+            observable_data["value"] = self._job.observable_name
+
+        return observable_data
+
+    @property
+    @cache_memoize(timeout=sys.maxsize)
+    def organization_id(self) -> str:  # called only once (cached indefinitely)
+        # Create author (if not exists)
+        org = pycti.Identity(self.opencti_instance).create(
+            type="Organization",
+            name="IntelOwl",
+            description=(
+                "Intel Owl is an Open Source Intelligence, or OSINT solution"
+                " to get threat intelligence data about a specific file, an IP or a"
+                " domain from a single API at scale. [Visit the project on GitHub]"
+                "(https://github.com/intelowlproject/IntelOwl/)"
+            ),
+        )
+        return org["id"]
+
+    @property
+    @cache_memoize(timeout=sys.maxsize)
+    def marking_definition_id(self) -> str:  # called only once (cached indefinitely)
+        # Create the marking definition (if not exists)
+        md = pycti.MarkingDefinition(self.opencti_instance).create(
+            definition_type="TLP",
+            definition=f"TLP:{self.tlp['type'].upper()}",
+            x_opencti_color=self.tlp["color"].upper(),
+            x_opencti_order=self.tlp["x_opencti_order"],
+        )
+        return md["id"]
```

Comment:
Caching indefinitely can lead to various bugs that will be super hard to debug. The `pycti.Identity` call depends on the `self.opencti_instance` value so if user were to change the URL, things will end up breaking. I'll go through the pycti docs and refer a better solution.

Reply:
Things will also break if they manually change something on the OpenCTI dashboard.

Reply:
Earlier, I thought we were caching only for a particular instance of a connector call else I'd have pointed this out at that moment only.

Reply:
Thanks for pointing this out, it slipped out of my mind.

Reply:
I'm reverting back to what I did earlier.